integration_tests/test_domestic_migration.py

```diff
@@ -123,3 +124,66 @@ def test_only_living_people_move(simulants_on_adjacent_timesteps):
         movers = before["address_id"] != after["address_id"]
         assert after[movers]["tracked"].all()
         assert (after[movers]["alive"] == "alive").all()
+
+
+def test_address_ids_states_pumas(simulants_on_adjacent_timesteps, sim) -> None:
+    """Check that address_ids are mapping correctly to states/pumas. Note that
+    this test does not distinguish between the different types of moves (eg
+    household moves, business moves, individual moves, etc)
+    """
+    state_puma_map = get_state_puma_map(
+        sim._data.artifact.load("population.households").reset_index()
+    )
+    for before, after in simulants_on_adjacent_timesteps:
+        mask_movers = before["address_id"] != after["address_id"]
+        # TODO: Check that *most* are in a new state when we add all locations
+        # Some movers are in a new state.
+        assert any(before.loc[mask_movers, "state"] != after.loc[mask_movers, "state"])
+
+        # Most movers are in a different puma
+        assert (
+            before.loc[mask_movers, "puma"] != after.loc[mask_movers, "puma"]
+        ).sum() / mask_movers.sum() >= 0.95
+
+        # puma/state do not change if address_id does not change
+        # (except for gq locations - those can have the same address_id and
+        # different states/pumas)
+        same_addresses = before[before["address_id"] == after["address_id"]].index
+        gq_simulants = after[after["housing_type"] != "Standard"].index
+        same_non_gq_addresses = same_addresses.difference(gq_simulants)
+
+        pd.testing.assert_frame_equal(
+            before.loc[same_non_gq_addresses, ["state", "puma"]],
+            after.loc[same_non_gq_addresses, ["state", "puma"]],
+        )
+
+        # Ensure pumas map to correct state. This will be an imperfect test
```

Comment:
The rest of this test could be on each timestep's population instead of being on simulants_on_adjacent_timesteps. There is an existing test for address_ids that is similar.